src/resources/assets/js/mixins/largoContainer.vue

```diff
@@ -176,15 +197,42 @@ export default {
         }
     },
     methods: {
-        getAnnotations(label) {
+        compileFilters(filters, union) {
+            let parameters = [];
+
+            filters.forEach(
+                (filter) => {
+                    //If we have an union in the current filter, we need to execute the last request
+                    if (!parameters[filter.filter]) {
+                        parameters[filter.filter] = [];
+                    }
+                    parameters[filter.filter].push(filter.value);
+                }
+            )
+            parameters['union'] = union;
+            return parameters
+        },
+        getAnnotations(label, filters, union) {
             let promise1;
             let promise2;
 
-            if (!this.annotationsCache.hasOwnProperty(label.id)) {
-                Vue.set(this.annotationsCache, label.id, []);
+            let labelFilters = {label: label.id};
+            if (filters.length > 0){
+                labelFilters['filters'] = filters;
+                labelFilters['union'] = union;
+            }
+
+            let cacheKey = JSON.stringify(labelFilters);
+
+            if (!this.annotationsCache.hasOwnProperty(cacheKey)) {
+                Vue.set(this.annotationsCache, cacheKey, []);
                 this.startLoading();
-                promise1 = this.queryAnnotations(label)
-                    .then((response) => this.gotAnnotations(label, response), handleErrorResponse)
+                let requestParams = this.compileFilters(filters, union);
+                promise1 = this.queryAnnotations(label, requestParams).then(
+                    (response) => this.gotAnnotations(label, response),
+                    handleErrorResponse
+                ).then(a => Vue.set(this.annotationsCache, cacheKey, a))
+                .finally(this.finishLoading);
```

Comment:
This looks a little cramped. What about:
```suggestion
                promise1 = this.queryAnnotations(label, requestParams)
                    .then(r => this.gotAnnotations(label, r), handleErrorResponse)
                    .then(a => Vue.set(this.annotationsCache, cacheKey, a))
                    .finally(this.finishLoading);
```